pkg/rollout/rollout_test.go

```diff
@@ -148,13 +159,43 @@ func TestUpdateService(t *testing.T) {
 			outAnnotations: map[string]string{
 				rollout.StableRevisionAnnotation:    "test-001",
 				rollout.CandidateRevisionAnnotation: "test-002",
+				rollout.LastRollForwardAnnotation:   clockMock.Now().Format(time.RFC3339),
 			},
 			outTraffic: []*run.TrafficTarget{
 				{RevisionName: "test-001", Percent: 100 - strategy.Steps[2], Tag: rollout.StableTag},
 				{RevisionName: "test-002", Percent: strategy.Steps[2], Tag: rollout.CandidateTag},
 				{LatestRevision: true, Tag: rollout.LatestTag},
 			},
 		},
+		// Candidate is the same as before (and healthy), but not enough time
+		// since last roll forward has passed.
+		{
+			name: "no enough time, do not roll forward",
+			traffic: []*run.TrafficTarget{
+				{RevisionName: "test-001", Percent: 100 - strategy.Steps[1], Tag: rollout.StableTag},
+				{RevisionName: "test-002", Percent: strategy.Steps[1], Tag: rollout.CandidateTag},
+				{LatestRevision: true, Tag: rollout.LatestTag},
+			},
+			annotations: map[string]string{
+				rollout.LastRollForwardAnnotation: clockMock.Now().Format(time.RFC3339),
+			},
```

Comment:
create inline function


```
mkAnnotations := func(t time.Time, offset Duration) map[string]string { 
} 
```

inside the test function

Reply:
Since it is also used in combination with other annotations, I created an utility function to generate the value instead.